geo/src/main/java/com/tomtom/speedtools/geometry/GeoPolyLine.java

```diff
@@ -52,23 +52,19 @@ public GeoPolyLine(@Nonnull final List<GeoPoint> points) {
 
         // Check the need to correct the elevation in a list of points.
         final List<GeoPoint> elevatedPoints;
-        final boolean noneElevated = points.stream().allMatch(x -> (x.getElevationMeters() != null));
-        final boolean allElevated = points.stream().allMatch(x -> (x.getElevationMeters() == null));
+        final boolean noneElevated = points.stream().noneMatch(x -> (x.getElevationMeters() != null));
+        final boolean allElevated = points.stream().noneMatch(x -> (x.getElevationMeters() == null));
 
         if (allElevated || noneElevated) {
 
             // Don't need to create a new list.
             elevatedPoints = points;
         } else {
 
-            // Find the first elevation.
-            double elevationMeters = Double.NaN;
-            for (final GeoPoint point : points) {
-                if (point.getElevationMeters() != null) {
-                    elevationMeters = point.getElevationMetersOrNaN();
-                    break;
-                }
-            }
+            // Find the first elevation, or use NaN.
+            double elevationMeters = points.stream().filter(point -> point.getElevationMeters() != null).findFirst().
+                    orElse(new GeoPoint(0.0, 0.0, Double.NaN)).
```

Comment:
Because of the `allElevated` and `noneElevated` checks, you will always have a first point. Or do you need this `orElse` because of the Optional in the stream?